src/containerapp/azext_containerapp/containerapp_resiliency_decorator.py

```diff
@@ -0,0 +1,484 @@
+            timeout_def["connectionTimeoutInSeconds"] = self.get_argument_timeout_connection() if self.get_argument_timeout_connection() is not None else DEFAULT_CONNECTION_TIMEOUT
+
+        tcpretry_def = TcpRetryPolicyModel
+        if self.get_argument_tcp_retry_max_attempts() is not None:
+            tcpretry_def["maxConnectAttempts"] = self.get_argument_tcp_retry_max_attempts()
+
+        circuitbreaker_def = CircuitBreakerPolicyModel
+        if self.get_argument_cb_consecutive_errors() is not None or \
+           self.get_argument_cb_interval() is not None or \
+           self.get_argument_cb_max_ejection() is not None:
+            circuitbreaker_def["consecutiveErrors"] = self.get_argument_cb_consecutive_errors() if self.get_argument_cb_consecutive_errors() is not None else DEFAULT_CONSECUTIVE_ERRORS
+            circuitbreaker_def["intervalInSeconds"] = self.get_argument_cb_interval() if self.get_argument_cb_interval() is not None else DEFAULT_INTERVAL
+            circuitbreaker_def["maxEjectionPercent"] = self.get_argument_cb_max_ejection() if self.get_argument_cb_max_ejection() is not None else DEFAULT_MAX_EJECTION
+
+        tcp_connectionpool_def = TcpConnectionPoolModel
+        if self.get_argument_tcp_max_connections() is not None:
+            tcp_connectionpool_def["maxConnections"] = self.get_argument_tcp_max_connections()
+
+        http_connectionpool_def = HttpConnectionPoolModel
+        if self.get_argument_http1_max_pending_req() is not None or \
+           self.get_argument_http2_max_req() is not None:
+            http_connectionpool_def["http1MaxPendingRequests"] = self.get_argument_http1_max_pending_req() if self.get_argument_http1_max_pending_req() is not None else DEFAULT_HTTP1_MAX_PENDING_REQ
+            http_connectionpool_def["http2MaxRequests"] = self.get_argument_http2_max_req() if self.get_argument_http2_max_req() is not None else DEFAULT_HTTP2_MAX_REQ
+
+        self.containerapp_resiliency_def["properties"]["httpRetryPolicy"] = http_retry_def
+        self.containerapp_resiliency_def["properties"]["timeoutPolicy"] = timeout_def
+        self.containerapp_resiliency_def["properties"]["tcpRetryPolicy"] = tcpretry_def
+        self.containerapp_resiliency_def["properties"]["circuitBreakerPolicy"] = circuitbreaker_def
+        self.containerapp_resiliency_def["properties"]["tcpConnectionPool"] = tcp_connectionpool_def
+        self.containerapp_resiliency_def["properties"]["httpConnectionPool"] = http_connectionpool_def
+
+        self.containerapp_resiliency_def = clean_null_values(self.containerapp_resiliency_def)
+
+        if self.containerapp_resiliency_def is None or self.containerapp_resiliency_def == {}:
+            self.containerapp_resiliency_def["properties"] = {}
+    
+    def set_up_default_containerapp_resiliency(self):
```

Comment:
We include a circuit breaker in our default? And both http and tcp retries?

Reply:
Removed tcp retries, but yeah I was asked to add circuit breaker to our defaults.